### DISCUSSION TITLE
Dynamic resolution scaling

### DISCUSSION THREAD

**Discussion by Zod1234** [alvr-org/ALVR#1829] [General]
I've been using vrperfkit with good success to bump up the frame rates, but the problem is, if image scaling isn't built into the game, text legibility can suffer, with the only alternative is to back off the image scaler.

Dynamic resolution never seemed to catch on, but i think vr might be able to benefit from this.  I'm not sure if it's even possible to do with alvr or if it needs to be done in a game, but i thought if the frame rate is below a threshold or below refresh rate, scale back resolution until framerate achieved. And when the action is slow, just looking around or reading text, it would scale back up
 Could this theoretically be done in alvr or would it need to be done ingame?

**Comment by zmerp** [comment]
SteamVR already supports dynamic resolution scaling, but then ALVR up/downsamples it to the same resolution you configured at the start for video transcoding. It's possible to reconfigure the video size during streaming but it's not straight forward, and given that we have many separate encoder implementations, this quickly gets out of hand.

**Reply by Zod1234** [reply]
Huh, I never knew steamvr had dynamic resolution scaling.  That's too bad nonnative steamvr headsets can't normally use that.

What about if fsr like tech was integrated into alvr and calls were made to that scaling up or down as needed.  That leaves the output res as is and shouldn't affect the streaming video size.  I think anyway..

**Reply by zmerp** [reply]
There is no technical issue blocking this, we just haven't implemented this yet. And we plan to add Qualcomm superresolution on the client soon, no ETA.
